executor.go

```diff
@@ -275,6 +294,130 @@ func (e *Executor) provideEventLogger(t *Task) *Logger {
+			val = taskFlag.Default
+		}
+
+		flagValErrMsg := fmt.Sprintf("The type for this flag is `%s`. Please use `%s`", taskFlag.ValueType, flagTypeToGetter[taskFlag.ValueType])
+		flagArg := FlagArg{
+			// Return the raw string val passed.
+			Value: val,
+			BoolVal: func() bool {
+				if taskFlag.ValueType != "bool" {
+					panic(flagValErrMsg)
+				}
+
+				// If a bool flag is passed, it should always have a true value.
+				return true
+			},
+			IntVal: func() int {
+				if taskFlag.ValueType != "int" {
+					panic(flagValErrMsg)
+				}
+
+				// Support flags that are passed either like `--flag=1` or `--flag="1"`.
+				strippedVal := stripWrappingQuotations(val)
+				num, err := strconv.Atoi(strippedVal)
+				if err != nil {
+					panic(fmt.Sprintf("Please pass an integer as the value. Err: %s", err))
+				}
+				return num
+			},
+			StringVal: func() string {
+				if taskFlag.ValueType != "string" {
+					panic(flagValErrMsg)
+				}
+				// Support flags that are passed either like `--flag=val` or `--flag="val"`.
+				strippedVal := stripWrappingQuotations(val)
+				return strippedVal
+			},
+		}
+		taskFlagsMap[key] = flagArg
+
+		// Allow readers to index into flag map via either LongName or ShortName
+		// regardless of which arg was passed in if both names are available.
+		if len(key) == 1 && key != taskFlag.LongName && taskFlag.LongName != "" {
+			// If only one char was passed through, check whether a LongName is available
+			// and also register it in the map.
+			key = taskFlag.LongName
+			taskFlagsMap[key] = flagArg
+		} else if len(key) > 1 && key == taskFlag.LongName && taskFlag.ShortName != 0 {
+			// If >1 char was passed through, check whether a ShortName is available
+			// and also register it in the map.
+			key = string(taskFlag.ShortName)
+			taskFlagsMap[key] = flagArg
+		}
+	}
+
+	return taskFlagsMap
+}
+
+func stripWrappingQuotations(str string) string {
+	strippedVal := str
+	if len(str) >= 2 && strings.HasPrefix(str, "\"") && strings.HasSuffix(str, "\"") {
+		strippedVal = str[1 : len(str)-1]
+	}
+
+	return strippedVal
+}
+
+func (e *Executor) getVerifiedFlagKey(taskName string, flagKey string) (string, error) {
+	var strippedKey string
+	if strings.HasPrefix(flagKey, "--") {
+		// If the flag key is prefixed with "--", we expect it to be the flag LongName.
+		strippedKey = string(flagKey[2:])
+		if len(strippedKey) <= 1 {
+			return flagKey, errors.New(fmt.Sprintf("Unknown flag: `%s`", strippedKey))
+		}
+	} else if strings.HasPrefix(flagKey, "-") {
+		// If the flag key is prefixed with "-", we expect it to be the flag ShortName.
+		strippedKey = string(flagKey[1:])
+		if len(strippedKey) != 1 {
+			return flagKey, errors.New(fmt.Sprintf("Did you mean `--%s` (with two dashes)?", strippedKey))
+		}
+	} else {
+		// If the flag is not prefixed with any dashes, this is invalid syntax.
+		return flagKey, errors.New(fmt.Sprintf("Options must be specified with either `-` or `--`"))
```

Comment:
micronit: maybe this err message could specify `-` for short names and `--` for long names?

Reply:
Done 👍